content/blog/2021-02-01-tidynhl-analyse/index.fr.md

```diff
@@ -0,0 +1,710 @@
+## 6705:   8478916   Joey Daccord  20182019      2018-19     regular       9               OTT           NA           NA             NA            NA               NA         NA         NA            NA         NA         NA           NA             NA          NA              NA                NA               NA            NA              NA                NA               NA            NA              NA                NA               NA            NA            1              1           0             1           0         0               0                 40                  5      0.8750000   5.000000    60.0000                    35                     3         0.9142857                     3                     2         0.3333333                     2                     0                 1
+```
+
+Finalement, nous allons appliquer quelques transformations et manipulations à ces
+trois jeux de données afin de régler quelques détails techniques importants pour l'analyse.
+
+
+```r
+# On se crée une fonction pour aggréger les données essentielles
+# On va devoir aggréger plusieurs fois les memes stats (pts, games, wins) dans l'analyse
+aggregate_stats <- function(dt, old_by_names, new_by_names) {
+  
+  dt_aggregated <- dt[, .(
+    skater_points = sum(skater_points, na.rm = TRUE),
+    skater_games = sum(skater_games, na.rm = TRUE),
+    goalie_games = sum(goalie_games, na.rm = TRUE),
+    goalie_wins = sum(goalie_wins, na.rm = TRUE)
+  ), old_by_names]
+  
+  # On additionne les games des goalies et des players car stockés dans 2 variables différentes
+  dt_aggregated[, player_games := ifelse(is.na(skater_games), 0, skater_games) + ifelse(is.na(goalie_games), 0, goalie_games)]
+  setnames(dt_aggregated, old_by_names, new_by_names)
+  
+}
+
+# Fusionner les équipes déménagés
+dt_stats <- merge_moved_teams(dt_stats)
+
+# Aggrégé les données par équipe jouée et joindre les données
+dt_stats_aggregated <- aggregate_stats(
+  dt = dt_stats,
+  old_by_names = c("player_id", "team_abbreviation"), 
+  new_by_names = c("player_id", "team_played")
+)
+
+dt_all <- merge(
+  x = dt_all,
+  y = dt_stats_aggregated,
+  by = c("player_id"),
+  all.x = TRUE
+)
```

Comment:
Même commentaire que plus haut pour le left join. Mais tsé, ça fonctionne là, t'es pas obligé de changer, c'est surtout pour que tu saches que la go to syntax de `data.table` c'est `X[Y]` et non la fonction `merge()`

Reply:
Done